File: doma-core/src/main/java/org/seasar/doma/jdbc/criteria/declaration/DeleteDeclaration.java
package org.seasar.doma.jdbc.criteria.declaration;

import java.util.Objects;
import java.util.function.Consumer;
import org.seasar.doma.jdbc.criteria.context.DeleteContext;

public class DeleteDeclaration {

  private final DeleteContext context;

  public DeleteDeclaration(DeleteContext context) {
    this.context = Objects.requireNonNull(context);
  }

  public DeleteContext getContext() {
    return context;
  }

  public void where(Consumer<WhereDeclaration> block) {
    Objects.requireNonNull(block);
    WhereDeclaration declaration = new WhereDeclaration(context);
    block.accept(declaration);
  }
}

File: doma-criteria/src/main/kotlin/org/seasar/doma/criteria/declaration/DeleteDeclaration.kt
package org.seasar.doma.criteria.declaration

import org.seasar.doma.criteria.context.DeleteContext

class DeleteDeclaration(private val context: DeleteContext) {
    private val whereDeclaration = WhereDeclaration(context.config) { context.where.add(it) }

    fun where(block: WhereDeclaration.() -> Unit) = whereDeclaration.block()
}
